src/Message/IPNResponse.php

```diff
@@ -19,7 +19,22 @@ public function __construct(RequestInterface $request, $data) {
 
     protected function verifyHash() {
         $hash = $this->getRequest()->getHashString($this->data);
-        return $hash === $this->data['HASH'];
+        $match = $hash === $this->data['HASH'];
+        /**
+         * The docs do not explicitly state this, but it seems like they are decoding html entities
+         * before calculating the hash. This results in signature mismatch when the payload contains
+         * encoded html entities. Only trying this if the original match fails here to keep it backward compatible.
+         */
+        if (!$match && is_array($data)) {
```

Comment:
`$data` isn't defined.
```suggestion
        if (!$match && is_array($this->data)) {
```